src/expandableCalendar/WeekCalendar/index.tsx

```diff
@@ -149,74 +85,77 @@ class WeekCalendar extends Component<WeekCalendarProps, State> {
         key={item}
         current={item}
         firstDay={firstDay}
-        style={this.getWeekStyle(this.containerWidth, style)}
+        style={getWeekStyle}
         markedDates={markedDates}
         onDayPress={onDayPress}
         context={currentContext}
         numberOfDays={context.numberOfDays}
         timelineLeftInset={context.timelineLeftInset}
       />
     );
-  };
+  },[]);
 
-  getItemLayout = (_: any, index: number) => {
-    return {
-      length: this.containerWidth,
-      offset: this.containerWidth * index,
-      index
-    };
-  };
-
-  keyExtractor = (_: string, index: number) => index.toString();
+  const keyExtractor = useCallback((_, index: number) => index.toString(), []);
 
-  renderWeekDaysNames = () => {
+  const renderWeekDaysNames = useMemo(() => {
     return (
-      <WeekDaysNames 
-        firstDay={this.props.firstDay} 
-        style={this.style.dayHeader} 
+      <WeekDaysNames
+        firstDay={firstDay}
+        style={dayHeader}
```

Comment:
We usually don't consider the default style as dynamic so once you created the style ref it shouldn't change, thus the `dayHeader` should stay `style?.current?.dayHeader` and be removed from the dependencies array.